File: app/src/main/java/net/ducksmanager/whattheduck/PublicationAdapter.java
package net.ducksmanager.whattheduck;

import android.app.Activity;
import android.content.Intent;
import android.view.View;

import net.ducksmanager.persistence.models.composite.InducksPublicationWithPossession;

import java.util.List;

import androidx.recyclerview.widget.RecyclerView;

import static net.ducksmanager.whattheduck.WhatTheDuck.selectedPublication;

public class PublicationAdapter extends ItemAdapter<InducksPublicationWithPossession> {

    PublicationAdapter(ItemList itemList, List<InducksPublicationWithPossession> items) {
        super(itemList, R.layout.row, items);
    }

    @Override
    protected ViewHolder getViewHolder(View v) {
        return new ViewHolder(v);
    }

    class ViewHolder extends ItemAdapter.ViewHolder {
        ViewHolder(View v) {
            super(v);
        }
    }

    @Override
    protected boolean isPossessed(InducksPublicationWithPossession inducksPublicationWithPossession) {
        return inducksPublicationWithPossession.isPossessed();
    }

    @Override
    protected View.OnClickListener getOnClickListener() {
        return view -> {
            int position = ((RecyclerView)view.getParent()).getChildLayoutPosition(view);
            selectedPublication = PublicationAdapter.this.getItem(position).publication.getPublicationCode();

            Intent i = new Intent(originActivity, IssueList.class);
            originActivity.startActivity(i);
        };
    }

    @Override
    protected Integer getPrefixImageResource(InducksPublicationWithPossession i, Activity activity) {
        return null;
    }


    @Override
    protected Integer getSuffixImageResource(InducksPublicationWithPossession i) {
        return null;
    }

    @Override
    protected String getIdentifier(InducksPublicationWithPossession i) {
        return i.publication.getPublicationCode();
    }

    @Override
    protected String getSuffixText(InducksPublicationWithPossession i) {
        return null;
    }

    @Override
    protected String getText(InducksPublicationWithPossession i) {
        return i.publication.getTitle();
    }

    @Override
    protected String getComparatorText(InducksPublicationWithPossession i) {
        return getText(i);
    }
}

File: app/src/main/java/net/ducksmanager/whattheduck/PublicationAdapter.kt
package net.ducksmanager.whattheduck

import android.app.Activity
import android.content.Intent
import android.view.View
import androidx.recyclerview.widget.RecyclerView
import net.ducksmanager.persistence.models.composite.InducksPublicationWithPossession

class PublicationAdapter internal constructor(
    itemList: ItemList<InducksPublicationWithPossession>,
    items: List<InducksPublicationWithPossession>
) : ItemAdapter<InducksPublicationWithPossession>(itemList, R.layout.row, items) {
    override fun getViewHolder(v: View?) = ViewHolder(v)

    inner class ViewHolder(v: View?) : ItemAdapter<InducksPublicationWithPossession>.ViewHolder(v!!)

    override fun isPossessed(item: InducksPublicationWithPossession): Boolean {
        return item.isPossessed
    }

    override val onClickListener: View.OnClickListener?
        get() = View.OnClickListener { view: View ->
            val position = (view.parent as RecyclerView).getChildLayoutPosition(view)
            WhatTheDuck.selectedPublication = getItem(position).publication.publicationCode
            originActivity.startActivity(Intent(originActivity, IssueList::class.java))
        }

    override fun getPrefixImageResource(i: InducksPublicationWithPossession, activity: Activity): Int? {
        return null
    }

    override fun getSuffixImageResource(i: InducksPublicationWithPossession): Int? {
        return null
    }

    override fun getIdentifier(i: InducksPublicationWithPossession): String? {
        return i.publication.publicationCode
    }

    override fun getSuffixText(i: InducksPublicationWithPossession): String? {
        return null
    }

    override fun getText(i: InducksPublicationWithPossession): String? {
        return i.publication.title
    }

    override fun getComparatorText(i: InducksPublicationWithPossession): String? {
        return getText(i)
    }
}